### PR TITLE
Improve the error messages displayed for hex encoding.

### PR DESCRIPTION
**Pull request by gavinbaumanis**
After a discussion on SLACK, here is a PR for improving the error message in HexCoder.java

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,8 @@ WEB-INF
 *.jfr
 
 /.vscode
+.vscode/
```
Comment (zspitzer) [inline .gitignore]: please remove this

**Review by zspitzer — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (gavinbaumanis) [reply .gitignore]: Hi @zspitzer, 
Done.

**Comment by zspitzer** [comment]
what do you think about including the invalid hex string in the error messages?

makes debugging after the fact a lot easier

**Comment by gavinbaumanis** [comment]
Added the argument's value into the exception thrown.

**Comment by zspitzer** [comment]
@gavinbaumanis almost there, the Lucee style for expections is to wrap all variable content in brackets, i.e `[ hexstring ]` which enables easy parsing of exception messages

**Comment by gavinbaumanis** [comment]
Thanks @zspitzer for helping me get this over line.
I know it's simple - but it has also served as some learning for submitting "Lucee" PRs.

**Comment by zspitzer** [comment]
@gavinbaumanis my pleasure mate!

File: core/src/main/java/lucee/runtime/coder/HexCoder.java [hunk 2]
```diff
@@ -89,9 +89,9 @@ else if (c >= 'a' && c <= 'f') {
 			retorno = (((byte) c) - 87);
 		}
 		else {
-			throw new CoderException("invalid hexadecimal String");
+			throw new CoderException("invalid hexadecimal String for, [ " + hexa + " ]. Allowed characters are [0-9], [a-f], [A-F]");
```
Comment (zspitzer) [inline core/src/main/java/lucee/runtime/coder/HexCoder.java]: `c` not `hexa`

**Review by zspitzer — CHANGES_REQUESTED** [review]

File: core/src/main/java/lucee/runtime/coder/HexCoder.java (hunk 2 above)
Comment (gavinbaumanis) [reply core/src/main/java/lucee/runtime/coder/HexCoder.java]: I should have stayed retired! Fixed. Thanks.

Comment (zspitzer) [reply core/src/main/java/lucee/runtime/coder/HexCoder.java]: building locally is really easy

https://docs.lucee.org/guides/working-with-source/build-from-source.html

**Review by zspitzer — APPROVED** [review]

**Comment by zspitzer** [comment]
just enabled a disabled test case which covers this function

https://github.com/lucee/Lucee/commit/6fbe7c1270cd790d83075bfe8ab9109613b3bd56

**Comment by zspitzer** [comment]
@gavinbaumanis congrats on your first commit to Lucee core!

**Comment by gavinbaumanis** [comment]
@zspitzer Thanks!

**Pull request merged**